lib/telemetry_metrics.ex

```diff
@@ -383,6 +383,11 @@ defmodule Telemetry.Metrics do
           reporter_options: reporter_options()
         }
 
+  @typedoc """
+  One of the base metric types.
+  """
+  @type metric() :: Counter.t() | LastValue.t() | Sum.t() | Summary.t() | Distribution.t()
```

Comment:
Should it be `t` to follow Elixir's convention?

Reply:
```suggestion
  @type t() :: Counter.t() | LastValue.t() | Sum.t() | Summary.t() | Distribution.t()
```

Reply:
Isn't `t()` always the module identity?

Reply:
No, it is the type represented by that module. Such as `Counter.t()` returns the Counter struct and so on. But it may be a union of existing types too!

Reply:
We already have `t()` as a generic "this is a metric definition". It's a map with all keys that reporters expect:

https://github.com/beam-telemetry/telemetry_metrics/blob/master/lib/telemetry_metrics.ex#L374-L384

Reply:
But we can switch and make it the other way around. I don't have a preference, and I think it's fine to swap them at this point.

Reply:
Ah, I see. Hrm, so why do we need `metric` if we already have this `t`?

Reply:
@bryannaegele Not the reporters themselves, I don't think so. The existing `t()` type is broader, so it could potentially break for those who pass something else than 5 base metric types to reporters. But at the same time I don't think there is any reporter that has a custom metric.

Reply:
@josevalim to have a short way of saying "I accept all base metric types", which, supposedly, is something that many reporters might want to say (StatsD and Prometheus do that). 

But yeah, maybe we don't need this and it's fine if each reporter just uses the union directly.

Reply:
If you support all, then I would go with t. Otherwise the manual union. So I think we can skip this? :)

Reply:
You can't really support all, because `t()` covers custom metrics. t() is more of a hint about what fields every metric has, so that you can implement a reporter on your own.

But we can ditch it, make t() a union, and write about metric fields in the "Writing reporters" guide.

Reply:
So I would make t to be the union. But then we are not allowing custom metrics, which may be fine.

Reply:
I was just looking for a convenience for reporters/docs so people don't have to put https://github.com/beam-telemetry/telemetry_metrics_prometheus_core/blob/master/lib/core.ex#L110-L117 everywhere.

I'm fine with skipping the union type in here if its an issue.

Reply:
@bryannaegele if the goal is to say what is supported, then it is best to explicitly list it. Otherwise we can add a new type, they may not support it, but if people use `.metric` or `.t` they always get the whole package.